underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsActiveSyncProvider.java

```diff
@@ -0,0 +1,197 @@
+  private void initNextWindow() {
+    for (String syncPoint : mActivity.keySet()) {
+      mActivity.put(syncPoint, new Integer(mActivity.get(syncPoint).intValue() / 10));
+      mAge.put(syncPoint, new Integer(mAge.get(syncPoint).intValue() + 1));
+    }
+  }
+
+  private void addFile(String syncPoint, String filePath) {
+    LOG.info("add file for syncPoint {}, at path {}", syncPoint, filePath);
+    AlluxioURI syncPointUri = new AlluxioURI(syncPoint);
+    if (!mChangedFiles.containsKey(syncPoint)){
+      mChangedFiles.put(syncPoint, new ConcurrentHashSet<>());
+      mActivity.put(syncPoint, new Integer(0));
+      mAge.put(syncPoint, new Integer(0));
+    }
+    mChangedFiles.get(syncPoint).add(
+        new AlluxioURI(syncPointUri.getScheme(), syncPointUri.getAuthority(), filePath));
+    mActivity.put(syncPoint, new Integer(mActivity.get(syncPoint).intValue() + 1));
+  }
+
+  private void clearFile(String syncPoint){
+    LOG.info("clear file for syncPoint {}, ", syncPoint );
+    mChangedFiles.remove(syncPoint);
+    mActivity.remove(syncPoint);
+    mAge.remove(syncPoint);
+  }
+
+  private boolean isEventModification(Event inotifyEvent) {
+    List<Event.EventType> eventList = Arrays.asList(new Event.EventType[]{Event.EventType.APPEND,
+        Event.EventType.CREATE, Event.EventType.METADATA, Event.EventType.RENAME,
+        Event.EventType.TRUNCATE, Event.EventType.UNLINK});
+    return eventList.contains(inotifyEvent.getEventType());
+  }
+
+  public boolean processEvent(Event event, List<AlluxioURI> syncPointList) {
+    boolean isModified = isEventModification(event);
+    boolean fileMatch = false;
+    String filePath = "";
+    String renameFilePath = "";
+    switch (event.getEventType()) {
+      case CREATE:
+        Event.CreateEvent createEvent = (Event.CreateEvent) event;
+        filePath = createEvent.getPath();
+        break;
+      case UNLINK:
+        Event.UnlinkEvent unlinkEvent = (Event.UnlinkEvent) event;
+        filePath = unlinkEvent.getPath();
+        break;
+      case APPEND:
+        Event.AppendEvent appendEvent = (Event.AppendEvent) event;
+        filePath = appendEvent.getPath();
+        break;
+      case RENAME:
+        Event.RenameEvent renameEvent = (Event.RenameEvent) event;
+        filePath = renameEvent.getSrcPath();
+        renameFilePath = renameEvent.getDstPath();
+        break;
+      case METADATA:
+        Event.MetadataUpdateEvent metadataUpdateEvent = (Event.MetadataUpdateEvent) event;
+        filePath = metadataUpdateEvent.getPath();
+        break;
+      default:
+        break;
+    }
+    if (filePath.isEmpty()) {
+      return false;
+    }
+    for (AlluxioURI syncPoint :  syncPointList) {
+      try {
+        LOG.info("file path in processEvent" + filePath);
+        LOG.info("syncPoint in processEvent" + syncPoint.getPath());
+        // find out if the changed file falls under one of the sync points
+        if (PathUtils.hasPrefix(filePath, syncPoint.getPath())) {
```

Comment:
NIT: if this was a fixed capacity list, then we could iterate and do the same processing on both paths, right?